WooCommerce/Classes/ViewRelated/Products/Add Product/FirstProductCreated/FirstProductCreatedView.swift

```diff
@@ -7,23 +7,25 @@ final class FirstProductCreatedHostingController: UIHostingController<FirstProdu
     ///
     private var shareProductCoordinator: ShareProductCoordinator?
 
-    init(product: Product,
+    init(productURL: URL,
+         productName: String,
          showShareProductButton: Bool) {
-        super.init(rootView: FirstProductCreatedView(showShareProductButton: showShareProductButton))
-        rootView.onSharingProduct = { [weak self] in
+        let viewModel = FirstProductCreatedViewModel(productURL: productURL,
+                                                     productName: productName,
+                                                     showShareProductButton: showShareProductButton)
+        super.init(rootView: FirstProductCreatedView(viewModel: viewModel))
+        viewModel.launchAISharingFlow = { [weak self] in
             guard let self,
-                  let navigationController = self.navigationController,
-                  let productURL = URL(string: product.permalink) else {
+                  let navigationController = self.navigationController else {
                 return
             }
 
-            let shareProductCoordinator = ShareProductCoordinator(productURL: productURL,
-                                                                  productName: product.name,
+            let shareProductCoordinator = ShareProductCoordinator(productURL: viewModel.productURL,
+                                                                  productName: viewModel.productName,
```

Comment:
Nit: we can use the injected `productURL` and `productName` here instead of the `viewModel`'s. That way we can make `launchAISharingFlow` a parameter in the view model's initializer.

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/9901/commits/8b3adc6a30b0a7348e4ffc66fa7765484268b713

But I still cannot set `launchAISharingFlow` before initializing the `FirstProductCreatedHostingController`, as we have to read `self` from the closure. Kindly let me know if there is another way to achieve this.

Reply:
You are right, since we need reference to `self`, it doesn't seem to be possible.